src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1796,8 +1796,16 @@ bool isRefEscape
                     PropertySymbol p => p.GetMethod ?? p.SetMethod,
                     _ => throw ExceptionUtilities.UnexpectedValue(symbol)
                 };
-                var thisParameter = method?.ThisParameter;
-                var refKind = thisParameter?.RefKind ?? RefKind.None;
+
+                var refKind = RefKind.None;
+                ParameterSymbol? thisParameter = null;
+                if (method is not null &&
+                    method.TryGetThisParameter(out thisParameter) &&
```

Comment:
>TryGetThisParameter [](http://example.com/codeflow?start=27&length=19)

This change seems like a necessary change, but should we also add an implementation of `TryGetThisParameter()` to `NativeIntegerMethodSymbol`?

Reply:
I'm unsure if we should or not. The documentation on the method doesn't give a lot of insight into when it should or should not be implemented. Looking through other implementations it was hard to discern a good meaning either.